bin/alias.ml

```diff
@@ -79,19 +79,46 @@ let dep_on_alias_rec_multi_contexts ~dir:src_dir ~name ~contexts =
   let* dir =
     Action_builder.of_memo (find_dir_specified_on_command_line ~dir:src_dir)
   in
-  let+ is_nonempty_list =
+  let* is_nonempty_list =
     Action_builder.all
       (List.map contexts ~f:(fun ctx ->
            Action_builder.dep_on_alias_rec name ctx dir))
   in
   let is_nonempty = List.exists is_nonempty_list ~f:Fun.id in
   if (not is_nonempty) && not (Dune_engine.Alias.is_standard name) then
-    User_error.raise
+    let+ load_dir =
+      Action_builder.all
+      @@ List.map contexts ~f:(fun ctx ->
+             let dir =
+               Source_tree.Dir.path dir
+               |> Path.Build.append_source (Context_name.build_dir ctx)
+               |> Path.build
+             in
+             Action_builder.of_memo @@ Load_rules.load_dir ~dir)
+    in
+    let hints =
+      let rec aliases t =
+        match (t : Load_rules.Loaded.t list) with
+        | Build build :: t ->
+          Alias.Name.Map.union
+            ~f:(fun _ a _ -> Some a)
+            build.aliases (aliases t)
+        | _ :: t -> aliases t
+        | _ -> Alias.Name.Map.empty
```

Comment:
I don't think this is right. You should use `--context` or default to make the suggestion.

Reply:
As discussed in chat, getting hold of `context_arg` from the CLI is a bit tricky here as I would have to thread it through unrelated places.